core/src/com/cloud/agent/resource/virtualnetwork/VirtualRoutingResource.java

```diff
@@ -374,9 +375,9 @@ private Answer execute(AggregationControlCommand cmd) {
                 FileConfigItem fileConfigItem = new FileConfigItem(VRScripts.CONFIG_CACHE_LOCATION, cfgFileName, sb.toString());
                 ScriptConfigItem scriptConfigItem = new ScriptConfigItem(VRScripts.VR_CFG, "-c " + VRScripts.CONFIG_CACHE_LOCATION + cfgFileName);
                 // 120s is the minimal timeout
-                int timeout = answerCounts * _eachTimeout;
-                if (timeout < 120) {
-                    timeout = 120;
+                Duration timeout = Duration.standardSeconds(answerCounts * _eachTimeout.getStandardSeconds());
+                if (timeout.getStandardSeconds() < 120L) {
```

Comment:
Please consider extract ``120L`` to a constant of type ``Duration`` to intent more clear and provide further unit conversion safety (e.g. MINIMUM_VR_RESTART_TIMEOUT).